gpytorch/functions/_inv_matmul.py

```diff
@@ -7,6 +7,10 @@
 
 
 def _solve(lazy_tsr, rhs):
+    from ..lazy import CholLazyTensor, TriangularLazyTensor
+
+    if isinstance(lazy_tsr, (CholLazyTensor, TriangularLazyTensor)):
```

Comment:
For the `TriangularLazyTensor` this is necessary (otherwise downstream we try to compute a Cholesky decomposition of a triangular matrix). For the `CholLazyTensor` this is something that should be done anyway (it in fact is, but that's hidden inside the overwritten `_solve` below). I guess this highlights a pretty fundamental change, namely that with `TriangularLazyTensor` we now have a lazy tensor that is not symmetric.